File: app/buildz/tmp/kapt3/stubs/debug/com/bits/bee/bpmc/data/data_source/remote/apiservices/SignUpApiService.java
package com.bits.bee.bpmc.data.data_source.remote.apiservices;

import java.lang.System;

/**
 * Created by aldi on 30/08/22.
 */
@kotlin.Metadata(mv = {1, 7, 1}, k = 1, d1 = {"\u0000\u001e\n\u0002\u0018\u0002\n\u0002\u0010\u0000\n\u0000\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\bf\u0018\u00002\u00020\u0001J \u0010\u0002\u001a\u000e\u0012\n\u0012\b\u0012\u0004\u0012\u00020\u00050\u00040\u00032\n\b\u0001\u0010\u0006\u001a\u0004\u0018\u00010\u0007H\'\u00a8\u0006\b"}, d2 = {"Lcom/bits/bee/bpmc/data/data_source/remote/apiservices/SignUpApiService;", "", "postRegister", "Lkotlinx/coroutines/flow/Flow;", "Lcom/bits/bee/bpmc/utils/ApiResponse;", "Lcom/bits/bee/bpmc/data/data_source/remote/response/SignUpResponse;", "signUpPost", "Lcom/bits/bee/bpmc/data/data_source/remote/post/SignUpPost;", "app_debug"})
public abstract interface SignUpApiService {
    
    @org.jetbrains.annotations.NotNull()
    @retrofit2.http.POST(value = "/rest/trial/signup")
    public abstract kotlinx.coroutines.flow.Flow<com.bits.bee.bpmc.utils.ApiResponse<com.bits.bee.bpmc.data.data_source.remote.response.SignUpResponse>> postRegister(@org.jetbrains.annotations.Nullable()
    @retrofit2.http.Body()
    com.bits.bee.bpmc.data.data_source.remote.post.SignUpPost signUpPost);
}

File: app/src/main/java/com/bits/bee/bpmc/data/data_source/remote/apiservices/SignUpApiService.kt
package com.bits.bee.bpmc.data.data_source.remote.apiservices

import com.bits.bee.bpmc.data.data_source.remote.post.*
import com.bits.bee.bpmc.data.data_source.remote.response.*
import com.bits.bee.bpmc.utils.ApiResponse
import kotlinx.coroutines.flow.Flow
import okhttp3.RequestBody
import retrofit2.http.*


/**
 * Created by aldi on 30/08/22.
 */
interface SignUpApiService {

    @POST("/rest/trial/signup")
    fun postRegister(@Body signUpPost: SignUpPost?): Flow<ApiResponse<SignUpResponse>>

    @GET("/rest/trial/usaha")
    fun getBidangUsaha() : Flow<ApiResponse<BidangUsahaResponse>>

    @GET("/rest/trial/city")
    fun getCity() : Flow<ApiResponse<CityResponse>>

    @POST("/rest/trial/confirm-sms")
    fun postVerifSms(@Body verifsmsPost: VerifSmsPost): Flow<ApiResponse<VerifSmsResponse>>

    @POST("/rest/trial/send-otp")
    fun postSendOtp(@Body verifsmsPost: SendOtpPost): Flow<ApiResponse<SendOtpResponse>>

    @POST("/rest/trial/change-pass-with-authkey")
    fun postSetupPassword(@Body setupPasswordPost: SetupPasswordPost): Flow<ApiResponse<SetupPasswordResponse>>

    @POST("/rest/trial/check-db")
    fun postVerifDb(@Body verifDbPost: VerifDbPost): Flow<ApiResponse<CheckDbResponse>>

}
